crates/uv-resolver/src/resolver/mod.rs

```diff
@@ -201,6 +190,20 @@ pub struct Resolver<'a, Provider: ResolverProvider, InstalledPackages: Installed
     visited: SharedSet<PackageName>,
     reporter: Option<Arc<dyn Reporter>>,
     provider: Provider,
+    solver_state: Option<SolverState<'a>>,
+}
+
+struct SolverState<'a> {
```

Comment:
What's the thinking behind splitting this type out? Maybe some code comments would help here.